Supports ignore flag before or after punctuation.

dancingplatypus opened the pull request:
- `/* ImageEmbed:skip */` can appear either before or after punctuation (comma or a semicolon).  
  _I had introduced a breaking change before and that was probably not wise._
- "www.placehold.it" is no good for supplying consistent images, so the test was removed.  
  _Need a better remote site for this sort of thing._

dancingplatypus committed:
imageEmbed:skip can appear before or after a semicolon or comma.
commented out the remote url test.  It works, but the site that is used for testing isn't working right.

ehynds commented:
Hi, I'm still seeing broken unit tests all over the place and the example is still broken. When I run `grunt` from within the  examples/ dir the output is this:

```
body {
  /* local images */
  background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAyRpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYxIDY0LjE0MDk0OSwgMjAxMC8xMi8wNy0xMDo1NzowMSAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIiB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIgeG1sbnM6c3RSZWY9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZVJlZiMiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENTNS4xIE1hY2ludG9zaCIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDo2Q0ZDOEMyN0YxM0IxMUUxOTRCM0IzMDk3OTU0RDM0NiIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDo2Q0ZDOEMyOEYxM0IxMUUxOTRCM0IzMDk3OTU0RDM0NiI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOjZDRkM4QzI1RjEzQjExRTE5NEIzQjMwOTc5NTREMzQ2IiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjZDRkM4QzI2RjEzQjExRTE5NEIzQjMwOTc5NTREMzQ2Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+UmaspwAAAA9JREFUeNpiOHPmDECAAQAEzAJlNBi8xAAAAABJRU5ErkJggg==');

  /* existing base64 images will be skipped */
  background-image: ;

  /* skip images using a comment directive */
  background-image: url(../images/test.png); /*ImageEmbed:skip*/

  /* external images work too */
  background-image: ;
}
```

I don't think the problem here is placehold.it itself (although I agree, this needs to be changed. We can point to the images in this repo on github.com instead) but rather the fact that the task can no longer handle external images at all.

ehynds commented:
I accidentally deleted the comment I just posted - but I fixed the failing unit tests by replacing the hard-coded newline characters with `grunt.utils.linefeed`. It looks like you're on windows which uses different characters than osx/linux.

Also, I created a unit test for external images, and replaced all placehold.it instances with the image http://erichynds.com/i/10x10.png. It looks like the regex is tweaked because it can't process external images at all it seems.

Anyway, great work, we're getting there, just need to resolve this last issue and I'll tag a new version and publish to npm.

Make sure to merge in my changes into your local branch. Ours have diverged enough that I couldn't cleanly merge in your last pull request (I did it manually, though).

dancingplatypus commented:
Oops... I just saw this.  I initiated another pull request, but I'll merge
your changes and update this a little later.  So please don't think I'm
being insensitive.  :-)

Look at how I changed the remote image fetching... I'm using the 'request'
library, which provides redirection and https.  As it turns out, github's
raw hides behind exactly this.  I used the "buffer" lib because I'm a
little lazy at reassembling chunks on a download.  I could recode it to
remove the dependency.

Anyway... in my branch I have all of the tests working, as well as the
example directory.  I was forcing quotes into the url( ), which I've
reversed.  I think the decision to keep these is arbitrary, and it adds
extra characters to the css.  I will merge your stuff and hit you up, but I
have to take a break.  I drove myself batty trying to figure out why the
request library doesn't auto-assemble a binary for you.  I guess I know
why... but... grrr....

On Mon, Oct 8, 2012 at 10:26 AM, Eric Hynds notifications@github.comwrote:

> I accidentally deleted the comment I just posted - but I fixed the failing
> unit tests by replacing the hard-coded newline characters with
> grunt.utils.linefeed. It looks like you're on windows which uses
> different characters than osx/linux.
> 
> Also, I created a unit test for external images, and replaced all
> placehold.it instances with the image http://erichynds.com/i/10x10.png.
> It looks like the regex is tweaked because it can't process external images
> at all it seems.
> 
> Anyway, great work, we're getting there, just need to resolve this last
> issue and I'll tag a new version and publish to npm.
> 
> Make sure to merge in my changes into your local branch. Ours have
> diverged enough that I couldn't cleanly merge in your last pull request (I
> did it manually, though).
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/ehynds/grunt-image-embed/pull/2#issuecomment-9227433.

dancingplatypus commented:
Oh, btw... the encoding issue was mostly because the url fetching is done
asynchronously, and wasn't tied into grunt properly.  I've buckled to the
async gods and rewrite the helper to take advantage of async.whilst

On Tue, Oct 9, 2012 at 8:58 PM, DancingPlatypus
dancingplatypus@gmail.comwrote:

> Oops... I just saw this.  I initiated another pull request, but I'll merge
> your changes and update this a little later.  So please don't think I'm
> being insensitive.  :-)
> 
> Look at how I changed the remote image fetching... I'm using the 'request'
> library, which provides redirection and https.  As it turns out, github's
> raw hides behind exactly this.  I used the "buffer" lib because I'm a
> little lazy at reassembling chunks on a download.  I could recode it to
> remove the dependency.
> 
> Anyway... in my branch I have all of the tests working, as well as the
> example directory.  I was forcing quotes into the url( ), which I've
> reversed.  I think the decision to keep these is arbitrary, and it adds
> extra characters to the css.  I will merge your stuff and hit you up, but I
> have to take a break.  I drove myself batty trying to figure out why the
> request library doesn't auto-assemble a binary for you.  I guess I know
> why... but... grrr....
> 
> On Mon, Oct 8, 2012 at 10:26 AM, Eric Hynds notifications@github.comwrote:
> 
> > I accidentally deleted the comment I just posted - but I fixed the
> > failing unit tests by replacing the hard-coded newline characters with
> > grunt.utils.linefeed. It looks like you're on windows which uses
> > different characters than osx/linux.
> > 
> > Also, I created a unit test for external images, and replaced all
> > placehold.it instances with the image http://erichynds.com/i/10x10.png.
> > It looks like the regex is tweaked because it can't process external images
> > at all it seems.
> > 
> > Anyway, great work, we're getting there, just need to resolve this last
> > issue and I'll tag a new version and publish to npm.
> > 
> > Make sure to merge in my changes into your local branch. Ours have
> > diverged enough that I couldn't cleanly merge in your last pull request (I
> > did it manually, though).
> > 
> > —
> > Reply to this email directly or view it on GitHubhttps://github.com/ehynds/grunt-image-embed/pull/2#issuecomment-9227433.
